src/components/ContentSidebar/MetadataSidebar.js

```diff
@@ -212,23 +236,113 @@ class MetadataSidebar extends React.PureComponent<Props, State> {
      * @return {void}
      */
     onModification = (id: string, isDirty: boolean) => {
-        const { editors = [] }: State = this.state;
-        const index = editors.findIndex(({ instance }) => instance.id === id);
-        if (index === -1) {
+        const oldEditor = this.getEditor(id);
+        if (!oldEditor) {
             return;
         }
+        const newEditor = { ...oldEditor };
+        newEditor.isDirty = isDirty;
+        this.onSaveSuccessHandler(oldEditor, newEditor);
+    };
 
-        const editor = { ...editors[index] };
-        editor.isDirty = isDirty;
-        const clone = editors.slice(0);
-        clone.splice(index, 1, editor);
-        this.setState({ editors: clone });
+    /**
+     * Handles a failed metadata fetch
+     *
+     * @private
+     * @param {Error} e - API error
+     * @param {string} code - error code
+     * @return {void}
+     */
+    fetchMetadataErrorCallback = (e: ElementsXhrError, code: string) => {
+        this.commonErrorCallback(e, code, {
+            editors: undefined,
+            error: messages.sidebarMetadataFetchingErrorContent,
+            templates: undefined,
+        });
+    };
+
+    /**
+     * Handles a successful metadata fetch
+     *
+     * @param {Object} file - the box file
+     * @return {void}
+     */
+    fetchMetadataSuccessCallback = ({
+        editors,
+        templates,
+    }: {
+        editors: Array<MetadataEditor>,
+        templates: Array<MetadataEditorTemplate>,
+    }) => {
+        this.setState({
+            editors: editors.slice(0),
+            error: undefined,
+            isLoading: false,
+            templates,
+        });
+    };
+
+    /**
+     * Fetches the metadata editors
+     *
+     * @return {void}
+     */
+    fetchMetadata(): void {
+        const { api, isFeatureEnabled = true }: Props = this.props;
```

Comment:
It seems like we should rely on `defaultProps` for `isFeatureEnabled` to avoid duplication.